common/src/main/java/io/druid/collections/IntList.java

```diff
@@ -112,7 +112,7 @@ public IntBuffer getBaseList(int index)
     final IntBuffer retVal = IntBuffer.wrap(array);
 
     if (index + 1 == baseListCount()) {
-      retVal.limit(maxIndex - (index * allocateSize));
+      retVal.limit(maxIndex - (index * allocateSize) + 1);
```

Comment:
It's a fix to a problem found with a unit test.  Check out the branch, remove the +1 and run the tests...